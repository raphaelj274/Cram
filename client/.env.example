# Use your IPv4 address here instead of localhost to be able to launch the app
# on your phone instead of on an emulator.
# Run `ipconfig getifaddr en0` in your terminal to get the IP address
# of your local network.
APOLLOCLIENT_HOST=
# Make the port number match the port used in server/src/index.ts
APOLLOCLIENT_PORT=4000