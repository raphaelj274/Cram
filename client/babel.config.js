module.exports = function (api) {
  console.log('babel.config.js called');
  api.cache(true);
  return {
    presets: ["babel-preset-expo"],
    // transform: {
      // "^.+\\.js$": "<rootDir>/node_modules/react-native/jest/preprocessor.js",
    //   "^.+\\.tsx?$": "ts-jest",
      // "^.+\\.[jt]sx?$": "babel-jest",
    }
  //   env: {
  //     "test": {
  //       "plugins": ["@babel/plugin-transform-modules-commonjs"]
  //     }
};



