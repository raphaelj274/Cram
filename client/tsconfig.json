{
  "extends": "expo/tsconfig.base",
  "compilerOptions": {
    "jsx": "react",
    "strict": true,
    // "allowSyntheticDefaultImports": true,
    // "jsx": "react-native",
    // "lib": ["dom", "esnext"],
    // "moduleResolution": "node",
    // "noEmit": true,
    // "skipLibCheck": true,
    // "resolveJsonModule": true,
    // "esModuleInterop": true
  }
}
