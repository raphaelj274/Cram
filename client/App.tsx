import { createStackNavigator } from '@react-navigation/stack';
import React from 'react';
import { Home } from './Components/Home';
import { Cram } from './Components/Cram';
import { Crammed } from './Components/Crammed';
import { NavigationContainer } from '@react-navigation/native';
import { ApolloProvider } from '@apollo/client/react';
import { apolloClient } from './Services/ApiService';
import Topic from './Interfaces/Topic';
import HeaderBtn from './Components/HeaderBtn';

type RootParamList = {
  Home: { paramA:  string }
  Cram: { paramB: string }
  Crammed: { paramC: Topic }
}

const Root = createStackNavigator<RootParamList>();

export default function App() {
  return (
    <ApolloProvider client={apolloClient}>
    <NavigationContainer>
      <Root.Navigator initialRouteName="Home" >
        <Root.Screen name="Home" component={Home} options={{ headerShown: false }}/>
        <Root.Screen name="Cram" component={Cram} options={{ headerRight: () => <HeaderBtn/> }}/>
        <Root.Screen name="Crammed" component={Crammed} options={{ headerRight: () => <HeaderBtn/> }} />
      </Root.Navigator>
    </NavigationContainer>
    </ApolloProvider>
  )
}