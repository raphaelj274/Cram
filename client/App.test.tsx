import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  fireEvent,
  waitFor,
  act,
  cleanup,
} from '@testing-library/react-native';
import { MockedProvider } from '@apollo/client/testing';
import fetchMock from 'jest-fetch-mock';
import 'react-native-gesture-handler/jestSetup';
import { AppWrapper } from './App';
import { Camera } from 'expo-camera';

//Warnings for animations
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');

  // The mock for `call` immediately calls the callback which is incorrect
  // So we override it with a no-op
  Reanimated.default.call = () => {};

  return Reanimated;
});

jest.mock('expo-camera', () => {
  const expoCamera: any = jest.createMockFromModule('expo-camera');
  expoCamera.Camera.requestPermissionsAsync = jest
    .fn()
    .mockResolvedValue({ status: 'granted' });
  return expoCamera;
});

describe('<App />', () => {
  afterEach(cleanup);

  it('placeholder test', () => {
    expect(1).toBe(1);
  });

  //INTEGRATION TEST
  it('after login, navigate to the cram page and render important components', async () => {
    fetchMock.enableMocks();

    const {
      getByPlaceholderText,
      getByTestId,
      getByText,
      findByText,
      findByTestId,
      debug,
    } = render(
      <MockedProvider>
        <AppWrapper />
      </MockedProvider>
    );

    // press login to use form
    act(() => {
      fireEvent.press(getByText('Log In'));
    });
    const loginBtn = getByTestId('arrow-btn');

    // fill out form and login
    const username = getByPlaceholderText('Username');
    const password = getByPlaceholderText('Password');
    act(() => {
      fireEvent.changeText(username, 'hello');
      fireEvent.changeText(password, 'world');
    });

    act(() => {
      fireEvent.press(loginBtn);
    });
    await waitFor(async () => {
      // Not checking for "...what do you need to cram?" message since it's
      // inside of Camera component which we're mocking
      await findByText('Cram');
      await findByText('Logout');
      await findByText('Home');
      await findByTestId('take-picture-btn');
      await findByTestId('CRAM-button');
    });
  });
});

//INTEGRATION TEST
it('All important components should load in main page after login', async () => {
  fetchMock.enableMocks();

  const {
    getByPlaceholderText,
    getByTestId,
    getByText,
    findByText,
    findByTestId,
    debug,
  } = render(
    <MockedProvider>
      <AppWrapper />
    </MockedProvider>
  );

  // press login to use form
  act(() => {
    fireEvent.press(getByText('Log In'));
  });
  const loginBtn = getByTestId('arrow-btn');

  // fill out form and login
  const username = getByPlaceholderText('Username');
  const password = getByPlaceholderText('Password');
  act(() => {
    fireEvent.changeText(username, 'hello');
    fireEvent.changeText(password, 'world');
  });

  await act(() => {
    fireEvent.press(loginBtn);
  });
  await waitFor(() => {
    // this tests main.jsx layout after login
    findByText('Hi hello, what do you need to cram?');
    findByText('CRAM');
    findByText('Home');
    findByText('Logout');
    findByTestId('take-picture-btn');
    findByTestId('expo-camera');
    const tree = renderer.create(<Camera />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
