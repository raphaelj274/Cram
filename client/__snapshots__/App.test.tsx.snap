// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All important components should load in main page after login 1`] = `null`;
